test/core/credentials/transport/tls/spiffe_utils_test.cc

```diff
@@ -29,11 +30,58 @@
 #include "absl/strings/string_view.h"
 #include "gmock/gmock.h"
 #include "gtest/gtest.h"
+#include "src/core/util/json/json_object_loader.h"
+#include "src/core/util/json/json_reader.h"
+#include "src/core/util/load_file.h"
 #include "test/core/test_util/test_config.h"
+#include "test/core/test_util/tls_utils.h"
 
 namespace grpc_core {
 namespace testing {
 
+namespace {
+
+constexpr absl::string_view kCertificatePrefix =
+    "-----BEGIN CERTIFICATE-----\n";
+constexpr absl::string_view kCertificateSuffix =
+    "\n-----END CERTIFICATE-----\n";
+
+absl::StatusOr<X509*> ReadCertificate(absl::string_view raw_cert) {
```

Comment:
Same here, can we use the existing `ParsePemCertificateChain` method?

Reply:
Why do we need to add the header and footer? Shouldn't it work without it?

Reply:
I believe the underlying OpenSSL function requires a well-formed PEM certificate which is defined by this header and footer with the base64 encoded DER between them. And the content of the SPIFFE Bundle x5c field is just the base64 encoding itself.